lib/server.coffee

```diff
@@ -335,13 +331,19 @@ module.exports = exports = (argv) ->
 # Plugins are located in packages in argv.packageDir, with package names of the form wiki-plugin-*
     glob path.join(argv.packageDir, 'wiki-plugin-*', 'factory.json'), (e, files) ->
       if e then return res.e(e)
-      files = files.map (file) ->
-        return fs.createReadStream(file).on('error', res.e).pipe(JSONStream.parse())
 
-      es.concat.apply(null, files)
-        .on('error', res.e)
-        .pipe(JSONStream.stringify())
-        .pipe(res)
+      doFactories = (file, cb) ->
+        fs.readFile file, (err, data) ->
+          return cb() if err
+          try
+            factory = JSON.parse data
+            cb null, factory
+          catch err
+            return cb()
+
+      async.map files, doFactories, (e, factories) ->
+        res.e if e
```

Comment:
My coffeescript is a little rusty, but do we need parins here `res.e() if e`? Do we want to pass the error through from the fs like the old implementation, or do we want to hide that from the client? Seems like it should be reported somehow, maybe with console.error()?

Reply:
Yes, it would be a good idea to hide the fs error from the client, and log it on the server (with a time stamp). Also going to be worth looking at all the other errors messages, to ensure they are also getting reported on the server.

Reply:
it should have been `res.e(e) if e`